app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb

```diff
@@ -58,19 +56,53 @@ def raw_safe_delete_volume
     EmsRefresh.queue_refresh(ext_management_system)
   end
 
-  def self.params_for_create(provider)
-    services = provider.storage_services.map { |service| {:value => service.ems_ref, :label => service.name} }
+  def params_for_update
+    ems = ExtManagementSystem.find(ems_id)
+    storage_service_id = ems.cloud_volumes.find_by(:id => id)[:storage_service_id]
```

Comment:
The preferred way to reference an attribute on an object is with `.storage_service_id` rather than `[:storage_service_id]`

Reply:
Actually, `ems.cloud_volumes.find_by(:id => id)` is the same thing as `self` since you're in an instance method, so you just need to call `storage_service_id`.

Since `CloudVolume` has `belongs_to :storage_service` you can actually drop the explicity `find_by(:id => storage_service_id` and let rails do that query.

This whole section can become
```ruby
service_value = storage_service.name
```

Reply:
good suggestion. fixed